lib/http.js

```diff
@@ -0,0 +1,45 @@
+import { getLatestMetrics, followMetrics } from '../lib/metrics.js'
+import http from 'node:http'
+import { once } from 'node:events'
+
+const asyncHandler = async (req, res) => {
+  const url = new URL(req.url, 'http://localhost/')
+
+  if (url.pathname === '/metrics') {
+    res.setHeader('Content-Type', 'application/json')
+    if (url.searchParams.has('follow')) {
+      const controller = new AbortController()
+      const { signal } = controller
+      await Promise.all([
+        (async () => {
+          try {
+            await once(req, 'close')
+          } catch {
+          } finally {
+            controller.abort()
+          }
+        })(),
+        (async () => {
+          try {
+            for await (const line of followMetrics({ signal })) {
+              res.write(`${line}\n`)
+            }
+          } catch {}
+        })()
+      ])
```

Comment:
I would find this code easier to understand if you extracted out the two inner functions.

```js
// here
      await Promise.all([
        abortOnRequestClosed(req, controller),
        pipeMetricsToResponse(res)
      ]);

// later - some of this is reusable!
        async function abortOnRequestClosed(req, controller) {
          try {
            await once(req, 'close')
          } catch {
          } finally {
            controller.abort()
          }
        }

        async function pipeMetricsToResponse(res, signal) {
          try {
            for await (const line of followMetrics({ signal })) {
              res.write(`${line}\n`)
            }
          } catch {
            // FIXME: close the response 
          }
        }
```